zcash_client_backend/src/data_api.rs

```diff
@@ -804,39 +899,39 @@ impl<Nf, S> ScannedBlock<Nf, S> {
 ///
 /// The purpose of this struct is to permit atomic updates of the
 /// wallet database when transactions are successfully decrypted.
-pub struct DecryptedTransaction<'a> {
+pub struct DecryptedTransaction<'a, AccountId> {
```

Comment:
Requires a `CHANGELOG` entry, and so forth, I won't explicitly comment on the rest. Probably the best way to do this is to have a single changelog entry that describes the abstraction change and then notes all of the affected types.